### ISSUE TITLE
CUDA Out of Memory on RTX 5090 (32GB VRAM) across all generation modes (text2world, image2world, video2world)

### ISSUE THREAD

**Issue by XabierSener** [nvidia-cosmos/cosmos-predict2.5#135]
Hi team,

I am trying to use the Cosmos Predict 2.5 model across its different generation modes: text2world, image2world, and video2world.

My System Specifications:

System RAM: 128 GB

GPU: NVIDIA RTX 5090 (32 GB VRAM)

CUDA Version: 13.0

NVIDIA Driver: 580

The Issue:
I noticed in the documentation that the VRAM requirement for video2world is around 32.6GB. However, none of the three options are working for me. Regardless of the mode I choose, the process crashes and I consistently get a similar OOM message:

What I've tried:
I have played around with various CLI flags (reducing resolution, limiting frames, adjusting chunks, and using PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True) to lower the GPU VRAM load, but I always hit the exact same error mentioned above.

Question:
Is there any possibility, workaround, or specific combination of flags to run these models with my current 32GB VRAM hardware?

I would really appreciate any guidance or tips you could provide. I am relatively new to this ecosystem and find this project incredibly interesting.

Thanks in advance for your help!

Best regards,
Xabier

**Comment by ChangjingLiu** [comment]
I also have the same problem

**Comment by LiJunqiang10223492** [comment]
meet the same problem. 
OOM alse occurred even generate text2world with the parameter --model=2B/distilled.

Additionally, as there are two GPUs on my server. When I tried using the following command, both ranks encountered OOM.
torchrun --nproc_per_node=2

My System Specifications:
System RAM: 64 GB
2 GPU: NVIDIA RTX 4000 (20 GB VRAM)
CUDA Version: 13.2
NVIDIA Driver: 595

**Comment by Junye-Chen** [comment]
Do you use the 2B model? I met the same problems. @XabierSener

**Comment by Anastasiawd** [comment]
Just set `--offload-diffusion-model   --offload-text-encoder   --offload-tokenizer --disable-guardrails` and I successfully ran on a single RTX 4090 GPU (24GB)

**Comment by PiresFabio** [comment]
I have an rtx 6000 ada. I had the same problem. I used this to generate samples export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True
PYTHONPATH=$(pwd) python examples/inference.py \
  -i my_nurec_image2world.json \
  -o outputs/(Your_Path_Location)
  --inference-type=image2world \
  --model=2B/post-trained \
  --offload-diffusion-model \
  --offload-text-encoder \
  --offload-tokenizer \
  --resolution 512,512 \
  --num-output-frames 25

**Comment by yogeshpaul** [comment]
You can reduce resolution to 480p. I was able to get below command working on 3090 with 2B model.

```
python examples/inference.py \
  -i assets/base/robot_welding.json \
  -o outputs/robot_welding_low_mem \
  --inference-type=text2world \
  --offload-diffusion-model \
  --offload-tokenizer \
  --offload-text-encoder \
  --disable-guardrails \
  --num-output-frames 17 \
  --num-steps 20 \
  --resolution 640,480
```

**Comment by richardhoff88** [comment]
Hitting the same issue, specifically on action-conditioned video2world inference (examples/video2world_action.py)

I'm seeing
40GB A100: OOM
80GB A100: OOM even at --chunk_size 2 (2 output frames + 1 conditioning frame) — 480p, and guardrail and prompt refiner both disabled with --disable_guardrail --disable_prompt_refiner

2 things worth noting
1. --disable_prompt_refiner doesn't prevent loading the prompt refiner model. In examples/video2world_action.py's setup_pipeline(), load_prompt_refiner=True is hardcoded regardless of args.disable_prompt_refiner. Cosmos-Reason1-7B (7B params) still loads onto the GPU. Patching to load_prompt_refiner=not args.disable_prompt_refiner dropped process memory from 77.6GB to 68.5GB

2. The OOM happens inside the VAE encoder. Fails at tokenizers/tokenizer.py:50, x = torch.cat([cache_x, x], dim=2) inside the VAE's feat_cache mechanism. There is no spatial/temporal tiling option. 

Running this on A100-SXM4-80GB, CUDA 12.9, inside the nvcr.io/nvidia/cosmos/cosmos-predict2-container:1.0 Singularity image.
